src/libraries/System.Private.CoreLib/src/System/Environment.GetFolderPathCore.Unix.cs

```diff
@@ -119,8 +111,23 @@ private static string GetFolderPathCoreWithoutValidation(SpecialFolder folder)
                 case SpecialFolder.InternetCache:
                     return Path.Combine(home, "Library", "Caches");
 #else
+                case SpecialFolder.ApplicationData:
+                    return GetXdgConfig(home);
+                case SpecialFolder.LocalApplicationData:
+                    // "$XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored."
+                    // "If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used."
+                    string? data = GetEnvironmentVariable("XDG_DATA_HOME");
+                    if (string.IsNullOrEmpty(data) || data[0] != '/')
+                    {
+                        data = Path.Combine(home, ".local", "share");
+                    }
+                    return data;
+                case SpecialFolder.MyDocuments: // same value as Personal
+                    return ReadXdgDirectory(home, "XDG_DOCUMENTS_DIR", "Documents");
```

Comment:
There will be code that uses `MyDocuments` for `HOME`.
That code will break, and need to be updated to using `UserProfile` instead.

Reply:
With "that code" are you referring to code inside of .NET or user made programs?

Reply:
Both.